File: CPE-1star/UVA10038-Jolly Jumpers .py
#每兩個相差在1~n-1
#相差不能一樣
while True:
	try:
		n=list(map(int,input().split()))
		temp=[]
		check=0
		count=0
		for i in range(1,len(n)-1):
			temp.append(abs(n[i]-n[i+1]))
		for i in range(len(temp)):
			if temp[i]>(n[0]-1) or temp[i]<1:
				count+=1
			check=temp.count(temp[i])
			if check>1:
				count+=1
		if count!=0:
			print("Not jolly")
		else:
			print("Jolly")
	except EOFError:
		break

File: CPE-1star/UVA10038-Jolly Jumpers .cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    while (cin >> n) {
        vector<int> nums(n);
        for (int i=0;i<n;i++) {
            cin >> nums[i];
        }
        vector<int> temp(n-1);
        for (int i=1; i<n;i++) {
            temp[i-1]=abs(nums[i]-nums[i-1]);
        }
        sort(temp.begin(),temp.end());
        bool isJolly=true;
        for (int i=0; i<n-1;i++) {
            if (temp[i]!=i+1) {
                isJolly=false;
                break;
            }
        }
        if(isJolly){
            cout << "Jolly" << endl;
        } else {
            cout << "Not jolly" << endl;
        }
    }
    return 0;
}
